src/NLog/Internal/LogEventInfoXmlWriter.cs

```diff
@@ -0,0 +1,196 @@
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE 
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+// THE POSSIBILITY OF SUCH DAMAGE.
+// 
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Xml;
+
+namespace NLog.Internal
+{
+    /// <summary>
+    /// <see cref="XmlWriter"/> for <see cref="LogEventInfo"/>.
+    /// </summary>
+    public class LogEventInfoXmlWriter
+    {
+        /// <summary>
+        /// Writes the specified <paramref name="logEventInfo"/> to the <see cref="StringBuilder"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="StringBuilder"/> to write the xml to.</param>
+        /// <param name="logEventInfo">The log event information to write.</param>
+        public void Write(StringBuilder builder, LogEventInfo logEventInfo)
+        {
+            var settings = new XmlWriterSettings
+            {
+                Indent = true,
+                Encoding = Encoding.UTF8,
+                OmitXmlDeclaration = true
+            };
+
+            Write(builder, logEventInfo, settings);
+        }
+
+        /// <summary>
+        /// Writes the specified <paramref name="logEventInfo" /> to the <see cref="StringBuilder" />.
+        /// </summary>
+        /// <param name="builder">The <see cref="StringBuilder" /> to write the xml to.</param>
+        /// <param name="logEventInfo">The log event information to write.</param>
+        /// <param name="settings">The settings.</param>
+        public void Write(StringBuilder builder, LogEventInfo logEventInfo, XmlWriterSettings settings)
+        {
+            using (var xw = XmlWriter.Create(builder, settings))
+            {
+                Write(xw, logEventInfo);
+                xw.Flush();
+            }
+        }
+
+        /// <summary>
+        /// Writes the specified <paramref name="logEventInfo"/> to the <see cref="XmlWriter"/>.
+        /// </summary>
+        /// <param name="writer">The <see cref="XmlWriter"/> to write the xml to.</param>
+        /// <param name="logEventInfo">The log event information to write.</param>
+        public void Write(XmlWriter writer, LogEventInfo logEventInfo)
+        {
+            writer.WriteStartElement("LogEvent");
+
+            if (logEventInfo.SequenceID != default(int))
+                writer.WriteElementString("SequenceID", XmlConvert.ToString(logEventInfo.SequenceID));
```

Comment:
Do we need to pass the Culture Info in these XmlConvert.ToString? What do you think?